iphone/Maps/UI/Help/AboutController.swift

```diff
@@ -16,10 +16,28 @@ final class AboutController: MWMViewController, UITableViewDataSource, UITableVi
     df.timeStyle = .none
     return String(format: L("data_version"), df.string(from:mapsDate), mapsVersionInt)
   }
+  
+  private var onDidAppearCompletionHandler: (() -> Void)?
+
+  init(onDidAppearCompletionHandler: (() -> Void)? = nil) {
+    self.onDidAppearCompletionHandler = onDidAppearCompletionHandler
+    super.init(nibName: nil, bundle: nil)
+  }
+
+  @available(*, unavailable)
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
 
   private func isDonateEnabled() -> Bool {
     return Settings.donateUrl() != nil
   }
+  
+  override func viewDidAppear(_ animated: Bool) {
+    super.viewDidAppear(animated)
+    onDidAppearCompletionHandler?()
```

Comment:
nit: Is doing no-ops considered a normal practice for iOS? Why a more clear pattern is avoided?
```
if let handler = onDidAppearCompletionHandler {
  handler()
  onDidAppearCompletionHandler = nil
}
```

Reply:
Yes it is "optional chaining".
It is used to safely call optioanls without binding (just hide all this if-let bindings under the hood). If that's what you meant... I'll fix it to binding!


https://docs.swift.org/swift-book/documentation/the-swift-programming-language/optionalchaining/

Reply:
Yeah, that was clear. As a C++ performance engineer, I'm triggered by constructs like:
```
var i = nil
// later
i = nil
```
:)